### DISCUSSION TITLE
Does DITA-OT 3.4.1 support DITA-OT 1.7.2 or earlier versions

### DISCUSSION THREAD

**Discussion by ptiruvaipati** [dita-ot/dita-ot#4109] [Q&A]
Hi Team,

We are planning to upgrade DITA-OT from 1.7.2 to 3.4.1.

Does DITA-OT 3.4.1 support backward compatibility for 1.7.2 or earlier versions ?

Regards,
Pavan

**Comment by raducoravu** [comment]
Why DITA OT 3.4.1? Why not 3.7.1 or 4.0.1?
Ideally you should try to download and use DITA OT 3.4.1 with your project, see what problems you find.
Because it depends, if you did not customize the publishing in any way, things should work. 
If for example you have a PDF customization plugin or folder with various XSLT changes, you will probably need to work on those to make the customization compatible with a newer DITA OT version.
https://www.dita-ot.org/dev/topics/migration.html

**Reply by ptiruvaipati** [reply]
As per the information in the DITA-OT website(https://www.dita-ot.org/download), DITA OT 3.4.1 supports DITA 1.3 or earlier versions.

DITA-OT 3.7.1 or 4.0.1 supports DITA 2.0. 

DITA OT 3.4.1 uses FOP 2.4. Our application is also using fop 2.4.

We don't do any customization in the DITA-OT 3.4.1.

We want to upgrade to the DITA-OT version which supports backward compatibility for DITA-OT 1.7.2 or earlier versions.

Regards,
Pavan

**Reply by raducoravu** [reply]
So:

     DITA-OT 3.7.1 or 4.0.1 supports DITA 2.0.

They have experimental support for DITA 2.0 in addition to DITA 1.3 which they also support. So they should be more recent and better (contain more bug fixes) than DITA OT 3.4.1. 

      DITA OT 3.4.1 uses FOP 2.4. Our application is also using fop 2.4.

Apache FOP 2.6 (bundled with DITA OT 4.0.1 for example) is better (contains more bug fixes) than FOP 2.4. 

      We don't do any customization in the DITA-OT 3.4.1.

Good, then you can try to download for example the latest DITA OT 4.0.1 and see if it works with your content.

**Comment by ptiruvaipati** [comment]
I understand DITA OT 4.0.1 is the latest version and it's better than DITA OT 3.4.1.

Currently we are using DITA OT 1.7.2 which is 10 years old. 

If we upgrade to DITA OT 4.0.1, our application may not support some features and requires customization. 

Could you please confirm whether DITA OT 3.4.1 supports DITA 1.7.2 or earlier versions ?


Regards,
Pavan

**Reply by raducoravu** [reply]
Between any two DITA OT versions there may be incompatible changes, they are in general documented here:
https://www.dita-ot.org/dev/topics/migration.html
I cannot answer your question because it depends on your actual DITA XML content. This is why I recommend you to download and try using a newer DITA OT version with your actual content.
Here's how you can use the DITA Open Toolkit from the command line:
https://www.dita-ot.org/dev/topics/build-using-dita-command.html

About this question:

    Could you please confirm whether DITA OT 3.4.1 supports DITA 1.7.2 or earlier versions ?

It's like when you want to buy a newer car version:

    Could you please confirm whether car model X supports car model (X-1) or earlier versions ?

Car model X is a better car and has more features than car model (X-1). But it may not have exactly the same buttons in the same places or maybe car (X-1) had a very small particular feature that you were using and car model X no longer has it because it was considered not to be important. So you need to test drive the car.

**Comment by chrispy-snps** [comment]
@ptiruvaipati - I agree with @raducoravu. If you are going to take the time to try a _newer_ version, consider spending your time evaluating the _newest_ version. And even if you do find things that must be adjusted for the newest version, it is likely worth the effort to use the latest version and take advantage of its features and bug fixes.
